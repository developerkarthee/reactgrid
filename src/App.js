import React from 'react';
import './App.css';
import Shipment from './components/Shipment';
import {BrowserRouter, Route} from 'react-router-dom';
import ShipmentDetail from './components/ShipmentDetail';

function App() {
  
  return (
  <BrowserRouter>
   <main>
     <h2>Shippment information</h2>
     <section>
       {/* <Shipment></Shipment> */}
        <Route exact path="/" component={Shipment}></Route>
        <Route exact path="/:id" component={ShipmentDetail}></Route>
     </section>
     </main>
  </BrowserRouter>
  );
}

export default App;
