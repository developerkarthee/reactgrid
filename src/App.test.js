import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { shallow } from 'enzyme';
describe('App Component', ()=>{
  let mountedApp;
  beforeEach(()=>{
    mountedApp = shallow(<App />)
  });

  test('Testing the label shipment info is available', () => {
    const { getByText } = render(<App />);
    const linkElement = getByText(/Shippment information/i);
    expect(linkElement).toBeInTheDocument();
  });

  it('should render App Component layout', () =>{
    const mountedApp = shallow(<App />);
    let browserRouter = mountedApp.find('BrowserRouter');
    expect(mountedApp.getElements()).toMatchSnapshot();
    expect(browserRouter.length).toBe(1);
  }); 
})
