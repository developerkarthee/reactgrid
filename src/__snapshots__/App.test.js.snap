// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render App Component layout 1`] = `
Array [
  <BrowserRouter>
    <main>
      <h2>
        Shippment information
      </h2>
      <section>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/:id"
        />
      </section>
    </main>
  </BrowserRouter>,
]
`;
